source/02_algo/distributed_rl_eng.rst

```diff
@@ -0,0 +1,115 @@
+
+From the algorithm point of view, it is hoped to reduce the algorithm’s requirements for data throughput (such as tolerating older and more off-policy data) while ensuring the convergence of the algorithm, and improving the efficiency of data exploration and utilization of collected data (For example, modify the data sampling method, or combine some research related to data-efficiency in RL). So that it provides more space and possibilities for system design.
+
+To sum up, distributed reinforcement learning is a more comprehensive research subfield, which requires mutual perception and coordination of deep reinforcement learning algorithm + distributed system design.
+
+
+Research Direction
+---------
+
+System
+~~~~~~
+
+Overall Architecture
+^^^^^^^^^^
+For common decision problems, the two most commonly used distributed architectures are IMPALA [3]_ and SEED RL [4]_
+
+.. image:: ./images/impala.png
+  :align: center
+  
+- The former is the classic Actor-Learner mode; that is, the data collection and training sides are entirely separated, and the latest neural network model is regularly passed from the Learner to the Actor, and the Actor is sent to the Learner after collecting a certain amount of data (i.e. observations). If there are multiple Learners, they also periodically synchronize the gradients of the neural network (i.e. the data-parallel model in distributed deep learning).
+
+.. image:: ./images/seed_rl.png
+  :scale: 50%
+  :align: center
+
+- On the basis of the former, the latter is dedicated to optimizing the loss of the transmission model. SEED RL strips out the part used for inference to generate actions, and puts it together with the training end to update the model through efficient inter-TPU communication technology. The cost of passing the model in IMPALA is greatly reduced, and for cross-machine communication between the environment and the reasoning Actor, SEED RL uses an optimized gRPC scheme to pass the observation and action, so there is not much burden.
+
+.. note::
+There is no absolute superiority or inferiority between these two schemes. The key lies in the fact that for a practical decision-making problem, whether it is more expensive to transmit models across machines, or more expensive to transmit observation and action data across machines, if it is the former , and there are better communication components between GPU/TPU, then SEED RL is a better solution, if it is the latter, IMPALA is a more stable choice. In addition, IMPALA can accumulate a batch of data for data transmission, while SEED RL requires data transmission in each interactive frame. This is a classic data batch and stream processing comparison problem. For the current machine learning community, the former is generally more complex and user-friendly. Also, if the entire training procedure requires a higher degree of freedom and customization, such as dynamically controlling some behavior of the Actor, IMPALA is more convenient.
```

Comment:
- "The key lies i... stable choice." this sentence is too long, maybe splitting it into 2-3 sentences would make it more clear
- "more complex and user-friendly", this seems a contradiction, usually something complex is also less user friendly

Reply:
For the second comment: I know this is not logical, but that's what the original text says.

Reply:
For the first comment, would you mind helping me break it into small pieces, as I don't understand the logic behind the knowledge? I just translated the original text. Thanks a lot.

Reply:
There is no absolute superiority or inferiority between these two schemes. If it is more expensive to transmit models across machines and there are better communication components between GPU/TPU, then SEED RL could be a more suitable option. Otherwise, if it is more expensive to transmit observations and actions data across machines, then IMPALA is a more stable choice. In addition, IMPALA can accumulate a batch of data for data transmission, while SEED RL requires data transmission in each interactive frame. This is a classic data batch and stream processing comparison problem. For the current machine learning community, the former is generally more user-friendly. Also, if the entire training procedure requires a higher degree of freedom and customization, such as dynamically controlling some behavior of the Actor, IMPALA is more convenient.

Reply:
You can change with this, i have already solved the 2 comments in this new version

Reply:
Thanks a lot！！！：） Fixed√